### PR TITLE
SDP refactoring.

### PR DESCRIPTION
**Pull request by gegles**
* Ability to receive candidates in the offer_sdp
* Let libnice do most of the parsing (similar to librtcdc). PeerConnection only grabs the non ICE-related stuff.
* Push Json dependency outside of the lib. Unify offer/answer/candidates around SDP format (string) expected by libnice.
* Moved json lib to example code.
* Naming: Parse/Generate Offer/Answer at the PC level. Local/RemoteSDP at the ICE Level.
* Many other small tweaks.
* Testclient works fine.
* More improvements to come.

### CODE REVIEW

File: include/rtcdcpp/NiceWrapper.hpp [hunk 1]
```diff
@@ -37,8 +37,6 @@
 #include <memory>
 #include <string>
 
-#include <json/json.h>
```
Comment (chadnickbok) [inline include/rtcdcpp/NiceWrapper.hpp]: I get the changes; now that librtcdcpp is trying to match more closely with other WebRTC implementations, having this api accept JSON no longer makes sense.

File: include/rtcdcpp/PeerConnection.hpp [hunk 2]
```diff
@@ -60,25 +59,32 @@ class PeerConnection {
...
+   *
+   * Parse Offer SDP
+   */
+  void ParseOffer(std::string offer_sdp);
+  
+  /**
+   * Generate Answer SDP
    */
-  std::string GenerateSDP();
+  std::string GenerateAnswer();
```
Comment (chadnickbok) [inline include/rtcdcpp/PeerConnection.hpp]: I like this; and it makes it clear where to add "GenerateOffer" and "ParseAnswer" in the future.

File: src/NiceWrapper.cpp [hunk 3]
```diff
@@ -227,6 +237,21 @@ void NiceWrapper::Stop() {
   }
 }
 
+void NiceWrapper::ParseRemoteSDP(std::string remote_sdp) {
+  string crfree_remote_sdp = remote_sdp;
+
+  // TODO: Improve this. This is needed because otherwise libnice will wrongly take the '\r' as part of ice-ufrag/password.
```
Comment (chadnickbok) [inline src/NiceWrapper.cpp]: I hate this bug; and in general it makes me a little nervous about how stable libnice actually is.

File: src/NiceWrapper.cpp [hunk 4]
```diff
@@ -257,11 +282,7 @@ void NiceWrapper::SendLoop() {
   }
 }
 
-void NiceWrapper::SetRemoteCredentials(std::string username, std::string password) {
```
Comment (chadnickbok) [inline src/NiceWrapper.cpp]: Just so I'm clear - now we parse the entire SDP, and rely on libnice parsing to extract credentials?

If I remember correctly, there is a drawback to this approach, but I can't remember quite what it is right now.
This is great for now, but lets review in future.

File: src/NiceWrapper.cpp (hunk 3 above)
Comment (gegles) [inline src/NiceWrapper.cpp]: I hear you. I will dig into this a bit more at some point. It appears that libnice would rather deal with just "\n" everywhere, but maybe the "\r\n" issue only reveals itself for the ice-ufrag/password .... If that's the case, we can enforce that properly and convert "\r\n" to "\n" at the NiceWrapper level....

File: src/NiceWrapper.cpp (hunk 4 above)
Comment (gegles) [inline src/NiceWrapper.cpp]: Almost correct. ;-) The PC layer still has to extract what it needs (i.e. what is not ICE-specific), libnice then parses the offer/answer/candidate SDPs to get all the ICE-related stuff it needs.

For now, it appears easier/smarter for us to let libnice do its job, but totally open if things need to change.

File: include/rtcdcpp/NiceWrapper.hpp (hunk 1 above)
Comment (gegles) [inline include/rtcdcpp/NiceWrapper.hpp]: thx.

**Comment by gegles** [comment]
Thx Nick. This is just my first pass, I am planning to iterate over this a few more times. Right now, I am merely trying to get our stuff working with this (close I think).

**Pull request merged**